### PR TITLE
feat(ST-90): add endpoint for fetching user details

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change
-  add GET /auth/me
- Utilize existing JwtStrategy for user validation
- Implement in AuthController
- Add createUser decorators and jwt guard

## Associated Ticket Link(s)

- https://trello.com/c/0WjGORHp

## Screenshots / Screen Recordings
N/A

### CODE REVIEW

**Review by ragib-hossain-sazim — APPROVED** [review]

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -23,4 +26,10 @@ export class AuthController {
   login(@Body() loginDto: LoginDto): Promise<AuthResponseDto> {
     return this.authService.login(loginDto);
   }
+
+  @UseGuards(JwtAuthGuard)
+  @Get("me")
```
Comment (ragib-hossain-sazim) [inline src/auth/auth.controller.ts]: [JC] why are we naming the endpoint `me`? Can we not name it `current_user`?

Comment (ghost) [reply src/auth/auth.controller.ts]: I was just going with the provided template.

Comment (ishmam-iqbal-sazim) [inline src/auth/auth.controller.ts]: Any reason why this is in the auth controller?
I believe this was in the users controller in the template

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

File: src/auth/auth.controller.ts (hunk 1 above)
Comment (ghost) [reply src/auth/auth.controller.ts]: Yes bhaiya it was, as I was registering my user's in the auth, so I thought of fetching the user from the auth too.

Comment (ishmam-iqbal-sazim) [reply src/auth/auth.controller.ts]: This should be in the users controller.

**Review by ishmam-iqbal-sazim — APPROVED** [review]

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**